client/helpers.go

```diff
@@ -79,38 +80,34 @@ func changeTargetsDelegation(repo *tuf.Repo, c changelist.Change) error {
 		if err != nil {
 			return err
 		}
-		r, _, err := repo.GetDelegation(c.Scope())
-		if _, ok := err.(data.ErrNoSuchRole); err != nil && !ok {
-			// error that wasn't ErrNoSuchRole
+		_, err = repo.GetDelegationRole(c.Scope())
+		if _, ok := err.(data.ErrInvalidRole); err != nil && !ok {
+			// error that wasn't ErrInvalidRole
 			return err
 		}
+
 		if err == nil {
-			// role existed, attempt to merge paths and keys
-			if err := r.AddPaths(td.AddPaths); err != nil {
-				return err
-			}
-			return repo.UpdateDelegations(r, td.AddKeys)
-		}
-		// create brand new role
-		r, err = td.ToNewRole(c.Scope())
-		if err != nil {
-			return err
+			// role existed, merge paths and keys to this copy of the delegation role
+			addTD := changelist.TufDelegation{AddPaths: td.AddPaths, AddKeys: td.AddKeys}
+			return repo.UpdateDelegations(c.Scope(), addTD)
```

Comment:
I was debating whether to mention this before - it did seem like a little bit of an abstraction break for this to take a change object, and it be the ONLY place a change object is taken.  But having a huge list of arguments also seems kinda not great, so I am torn between the two.  One way to avoid having the changelist be imported here and to avoid taking a large number of arguments is to construct `tuf`'s own change object, and have there be a translation.  Or to have the `TufDelegation` object just be an embed of the `tuf` update-delegation object. :|  Which also isn't great, because then we have two objects that are essentially the same thing.

The other thing I was thinking was that maybe we want to eventually move all our `applyChange*` type functions into the `tuf`package, and maybe have that be the interface to update a TUF repo?  In which case maybe taking change objects would be less problematic?  (Although maybe we'd still want a layer in between, dunno)